newa/cli.py

```diff
@@ -231,30 +261,87 @@ def cmd_schedule(ctx: CLIContext) -> None:
         # generate all relevant test request using the recipe data
         # prepare a list of Request objects
 
+        # identify compose to be used
+        # just a dump conversion for now
+        compose = jira_job.erratum.release.rstrip('.GA') + '-Nightly'
+        initial_config = RawRecipeConfigDimension(compose=compose)
+
         config = RecipeConfig.from_yaml_url(jira_job.recipe.url)
+        # build requests
+        requests = config.build_requests(initial_config)
 
         # create few fake Issue objects for now
-        for request in config.build_requests():
-            request_job = RequestJob(
+        for request in requests:
+            schedule_job = ScheduleJob(
                 event=jira_job.event,
                 erratum=jira_job.erratum,
                 jira=jira_job.jira,
                 recipe=jira_job.recipe,
                 request=request)
-            ctx.save_request_job('request-', request_job)
+            ctx.save_schedule_job('schedule-', schedule_job)
 
 
 @main.command(name='execute')
+@click.option(
+    '--workers',
+    default=4,
+    )
 @click.pass_obj
-def cmd_execute(ctx: CLIContext) -> None:
+def cmd_execute(ctx: CLIContext, workers: int) -> None:
     ctx.enter_command('execute')
 
-    for erratum_job in ctx.load_erratum_jobs('schedule-'):
-        # worker = new Executor(yaml)
-        # run() returns result object
-        # result = worker.run()
+    if not os.environ.get("TESTING_FARM_API_TOKEN"):
+        raise ValueError("TESTING_FARM_API_TOKEN not set!")
+
+    # get a list of files to be scheduled so that they can be distributed across workers
+    schedule_list = [
+        os.path.join(
+            ctx.state_dirpath,
+            child.name)
```

Comment:
Please, stick to new and recommended https://docs.python.org/3/library/pathlib.html, `os.path` is supposed to be pushed out of our collective memory :)

```python
schedule_list = [
    ctx.state_dirpath / child.name
    for ...
```

Reply:
BTW a very helpful table to ease the transition: https://docs.python.org/3/library/pathlib.html#correspondence-to-tools-in-the-os-module

Reply:
fixed